src/sinks/elasticsearch/config.rs

```diff
@@ -237,7 +237,7 @@ impl ElasticsearchConfig {
     pub fn common_mode(&self) -> crate::Result<ElasticsearchCommonMode> {
         match self.mode {
             ElasticsearchMode::Bulk => Ok(ElasticsearchCommonMode::Bulk {
-                index: self.index().expect("index should not be undefined"),
+                index: self.index().ok_or("index should not be undefined")?,
```

Comment:
This will not panic but it will throw the Err upstream and will still result in the sink initialization erroring out.

From my PR which introduced the bug, I omitted this line in the index() function:

```
.unwrap_or_else(BulkConfig::default_index);
```

, which appears to be the root cause.

Could also experiment with setting the serde default in the top level config for `bulk` , to be explicitly "Some(Bulk:Config::default())" , since I think that is what I was trying to go for but since it's an Option, it is defaulting to None.

In either case, I would suggest running tests locally for this.

Reply:
I'd also suggest adding a test for this since we seem to be lacking coverage for configurations that don't include `bulk.index`.